### PR TITLE
pypi_formula_mappings: update for manim

### PR DESCRIPTION
**Pull request by chenrui333**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----
https://github.com/Homebrew/homebrew-core/actions/runs/9863188002/job/27237048710

### CODE REVIEW

File: Formula/m/manim.rb [hunk 1]
```diff
@@ -22,21 +22,34 @@ class Manim < Formula
...
+
   on_macos do
+    depends_on "gcc"
     depends_on "gettext"
     depends_on "harfbuzz"
+    depends_on "libimagequant"
+    depends_on "libraqm"
+    depends_on "libxcb"
+    depends_on "openjpeg"
+    depends_on "webp"
```
Comment (carlocab) [inline Formula/m/manim.rb:52]: You should be able to get rid of these if you remove the `pillow` resource.

File: Formula/m/manim.rb [hunk 2]
```diff
@@ -22,21 +22,34 @@ class Manim < Formula
...
   depends_on "pango"
   depends_on "pillow"
   depends_on "py3cairo"
   depends_on "python@3.12"
   depends_on "scipy"
 
+  uses_from_macos "zlib"
+
   on_macos do
+    depends_on "gcc"
```
Comment (carlocab) [inline Formula/m/manim.rb:45]: Might be able to get rid of this if you remove the `scipy` resource and use the formula instead.

File: Formula/m/manim.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/m/manim.rb:52]: cc @branchvincent for thoughts

Comment (branchvincent) [reply Formula/m/manim.rb:52]: sounds good to me, I think #171786 just missed updating `pypi_formula_mappings.json` (same for `scipy`)

**Review by branchv — APPROVED** [review]

**Pull request merged**